source/RunnersField.mc

```diff
@@ -51,11 +53,38 @@ class RunnersView extends Ui.DataField {
 	hidden var currentCadence = 0;
 	hidden var averageCadence = 0;
 	hidden var maxCadence = 0;
+	
+	hidden var hrZones = [ 122,134,149,163,182 ];
     
     hidden var hasBackgroundColorOption = false;
     
     function initialize() {
         DataField.initialize();
+        var str = Application.getApp().getProperty("hrZones");

```

Comment:
Could you extract that block as a method with a speaking method name that returns the zones, so that the code doesn't get too complex. Would that be ok for you?